### PR TITLE
sink(ticdc): add batch buffer for flush worker

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4423

### What is changed and how it works?
add batch buffer for flush worker. Avoid frequent memory allocation here.

### Check List 

Tests 

 - Unit test

Code changes

None

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/mq_flush_worker.go [hunk 1]
```diff
@@ -65,23 +65,28 @@ func newFlushWorker(encoder codec.EventBatchEncoder, producer producer.Producer,
 }
 
 // batch collects a batch of messages to be sent to the Kafka producer.
-func (w *flushWorker) batch(ctx context.Context) ([]mqEvent, error) {
-	var events []mqEvent
+func (w *flushWorker) batch(
+	ctx context.Context,
+	events []mqEvent,
+) (int, error) {
```
Comment (overvenus) [inline cdc/sink/mq_flush_worker.go:71]:
```suggestion
func (w *flushWorker) batch(
	ctx context.Context, events []mqEvent,
) (int, error) {
```

**Review by overvenus — APPROVED** [review]

File: cdc/sink/mq_flush_worker.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sink/mq_flush_worker.go:71]: Thanks for your review!  💚 💙 💜 💛 ❤️

**Review by sdojjy — APPROVED** [review]

**Pull request merged**